pkg/rancher-desktop/main/diagnostics/wslFromStore.ts

```diff
@@ -0,0 +1,221 @@
+      console.error(`Failed to check for WSL in store: ${ ex }`);
+
+      return 'There was an error checking if WSL was installed from the Microsoft Store.';
+    }
+  }
+
+  /**
+   * Run WSL and return stdout.
+   * @param args Command line for WSL
+   * @param failureMatch If given, ignore non-zero exit code if the output
+   *   contains this string.
+   */
+  async runWSL(args: string[], failureMatch?: string): Promise<string> {
+    try {
+      const result = await spawnFile(
+        path.join(this.system32, 'wsl.exe'), args,
+        { stdio: ['ignore', 'pipe', console], encoding: 'utf16le' });
+
+      return result.stdout;
+    } catch (ex: any) {
+      if (failureMatch && 'stdout' in ex) {
+        const stdout: string = ex.stdout;
+
+        if (stdout.includes(failureMatch)) {
+          return stdout;
+        }
+      }
+      throw ex;
+    }
+  }
+
+  _cachedWSLHelpText: string | undefined;
+  async getWSLHelpText(): Promise<string> {
+    if (this._cachedWSLHelpText) {
+      return this._cachedWSLHelpText;
+    }
+    // `wsl.exe --help` always exits with -1; that's okay as long as the output
+    // contains `wsl.exe` somewhere.
+    this._cachedWSLHelpText = await this.runWSL(['--help'], 'wsl.exe');
+
+    return this._cachedWSLHelpText;
+  }
+
+  async isKernelInstalled(): Promise<boolean> {
+    if ((await this.getWSLHelpText()).includes('--version')) {
+      const versions = await this.runWSL(['--version']);
+      const kernel = versions.split(/\r?\n/).map(s => s.trim()).find(v => /kernel/i.test(v));
+
+      console.debug(`Running WSL with ${ kernel || '(no kernel found)' }`);
+
+      return !!kernel;
+    } else {
+      try {
+        const status = await this.runWSL(['--status']);
+        const match = /:\s*((?:\d+\.){2,}\d+)/.exec(status);
+
+        if (match) {
+          console.debug(`Running WSL with kernel ${ match.groups?.[1] }`);
+
+          return true;
+        }
+        console.debug('`wsl --status` does not contain version string.');
+      } catch (ex) {
+        console.debug(ex);
+      }
+
+      return false;
+    }
+  }
+
+  async check() {
+    const storeURL = 'ms-windows-store://pdp/?ProductId=9P9TQF7MRM4R&mode=mini';
+
+    if (!(await this.getWSLHelpText()).includes('--exec')) {
+      // WSL is not installed
+      const description = 'Windows Subsystem for Linux is not installed.';
+      const fixes: { description: string, url?: string }[] = [];
+
+      if (await this.canUseStoreWSL()) {
+        fixes.push({
+          description: `Install Windows Subsystem for Linux from the [Microsoft Store](${ storeURL }).`,
+          url:         storeURL,
+        });
+      } else {
+        fixes.push({
+          description: `Install Windows Subsystem for Linux manually.`,
+          url:         'https://aka.ms/wslinstall',
+        });
+      }
+
+      return {
+        passed: false, description, fixes,
+      };
+    }
+
+    if (!await this.isKernelInstalled()) {
```

Comment:
Maybe I misremember this, but wasn't there also a requirement to have a certain minimum version of the kernel? Do we catch this here, if the kernel is installed, but too old?

Reply:
There's a minimum kernel version involved, but at this point I have no way to reproduce it in my environment so I can't really test it…